src/evaluators/state/PHAL_SaveSideSetStateField_Def.hpp

```diff
@@ -382,12 +441,12 @@ saveNodeState(typename Traits::EvalData workset)
     GO nodeId2d;
 
     // Loop on the sides of this sideSet that are in this workset
-    const std::vector<Albany::SideStruct>& sideSet = workset.sideSets->at(sideSetName);
-    for (auto const& it_side : sideSet)
+    sideSet = workset.sideSetViews->at(sideSetName);
+    for (int sideSet_idx = 0; sideSet_idx < sideSet.size; ++sideSet_idx)
     {
       // Get the data that corresponds to the side
-      const int cell     = it_side.elem_LID;
-      const int side     = it_side.side_local_id;
+      const int cell = sideSet.elem_LID(sideSet_idx);
+      const int side = sideSet.side_local_id(sideSet_idx);
 
       switch (dims.size())
```

Comment:
@mcarlson801 this `switch` needs to be modified for collapsed side sets (`dims.size()` is either 2 or 3), also when you access `dims`, you need to reduce the index by one.